lib/lower/convert-expr.cc

```diff
@@ -0,0 +1,921 @@
+//===-- lib/lower/convert-expr.cc -------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "convert-expr.h"
+#include "../common/default-kinds.h"
+#include "../common/unwrap.h"
+#include "../evaluate/fold.h"
+#include "../evaluate/real.h"
+#include "../semantics/expression.h"
+#include "../semantics/symbol.h"
+#include "../semantics/type.h"
+#include "bridge.h"
+#include "builder.h"
+#include "complex-handler.h"
+#include "convert-type.h"
+#include "mlir/Dialect/AffineOps/AffineOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/StandardOps/Ops.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/Passes.h"
+#include "optimizer/FIRDialect.h"
+#include "optimizer/FIROps.h"
+#include "optimizer/FIRType.h"
+#include "runtime.h"
+#include "llvm/ADT/APFloat.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/IR/Type.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace Br = Fortran::lower;
+namespace Co = Fortran::common;
+namespace Ev = Fortran::evaluate;
+namespace L = llvm;
+namespace M = mlir;
+namespace Pa = Fortran::parser;
+namespace Se = Fortran::semantics;
+
+using namespace Fortran;
+using namespace Fortran::lower;
+
+namespace {
+
+#define TODO()                                                                 \
+  assert(false);                                                               \
+  return {}
+
+/// Lowering of Fortran::evaluate::Expr<T> expressions
+class ExprLowering {
+  M::Location location;
+  AbstractConverter &converter;
+  M::OpBuilder &builder;
+  SomeExpr const &expr;
+  SymMap &symMap;
+#if 0
+  SymMap loadedSymbols;
+#endif
```

Comment:
Can we use a comment here or remove this code instead?

Reply:
Removed